src/chainstate/stacks/db/blocks.rs

```diff
@@ -4113,7 +4113,42 @@ impl StacksChainState {
                     let event = STXEventType::STXMintEvent(STXMintEventData { recipient, amount });
                     events.push(StacksTransactionEvent::STXEvent(event));
                 }
-                Ok((total_minted, events))
+
+                let tx_version = if mainnet {
+                    TransactionVersion::Mainnet
+                } else {
+                    TransactionVersion::Testnet
+                };
+                let boot_code_address = boot_code_addr(mainnet);
+                let boot_code_auth = TransactionAuth::Standard(
+                    TransactionSpendingCondition::Singlesig(SinglesigSpendingCondition {
+                        signer: boot_code_address.bytes.clone(),
+                        hash_mode: SinglesigHashMode::P2PKH,
+                        key_encoding: TransactionPublicKeyEncoding::Uncompressed,
+                        nonce: 0,
+                        tx_fee: 0,
+                        signature: MessageSignature::empty(),
+                    }),
+                );
+
+                let lockup_tx = StacksTransaction::new(
+                    tx_version,
+                    boot_code_auth,
+                    TransactionPayload::ContractCall(TransactionContractCall {
+                        address: boot_code_address,
+                        contract_name: ContractName::try_from("lockup").unwrap(),
+                        function_name: ClarityName::try_from("get-lockups").unwrap(),
+                        function_args: vec![block_height],
+                    }),
+                );
+                let lockup_receipt = StacksTransactionReceipt::from_stx_transfer(
+                    lockup_tx,
+                    events,
+                    Value::okay_true(),
+                    ExecutionCost::zero(),
+                );
+
+                Ok((total_minted, Some(lockup_receipt)))
```

Comment:
Why even include this in the event stream at all?  This information can be queried via the RESTful API already.

Reply:
It's so that an event listener can build up an account balance history that includes unlocks, and maybe just as importantly, it allows an event listener to accurately derive an account balance without needing to have a kind of split-brain operation where some things need to be queried using the RESTful API, whereas other derived values are computed by the event stream.

Reply:
That's part of why this is so confusing.  This `lockup_receipt` isn't actually returning any data -- it's just returning `(ok true)`, and isn't emitting any events of its own (`get-lockups` doesn't print anything or move any tokens).  So, none of the individual STX transfer events seem to be getting reported here.  Maybe that's a bug in this PR -- maybe the return value should be the `(list ..)` returned by `get-lockups`?

Reply:
It's getting associated with the StxMint events.

Reply:
Yeah, we are taking the events that were previously generated, and embedding them in a Receipt.
The reason for this is that the `stacks-blockchain-api` is not ingesting transactions events, it's ingesting transaction receipts.

Reply:
Ah, yes, I see that now.  Thanks for helping me understand!